bot.py

```diff
@@ -195,20 +195,29 @@ def check(m):
 
 	message = await bot.wait_for("message", check=check)
 	guess = message.content.split()[-1]
+	profile = await userprofile.load_profile(message.author.id)
 
-	while guess.lower() != name.lower():
-		if guess.lower() == "hint":
+	while True:
+		if profile is None:
+			await channel.send(
+				f"{message.author.mention} you need to run `p!begin` before you can play the game"
+			)
+		elif guess.lower() == "hint":
 			await channel.send(
 				f"The name of this **{util.type_to_string(pokemon.Type).pop()} type** pokemon starts with the letter **{name[0]}**"
 			)
 		elif Levenshtein.distance(guess.lower(), name.lower()) < 3:
 			await channel.send(
 				f"{message.author.mention} that guess was close, but not quite right"
 			)
+		elif guess.lower() == name.lower():
+			break
 		else:
 			await channel.send("That's incorrect, please guess again")
+
 		message = await bot.wait_for("message", check=check)
 		guess = message.content.split()[-1]
+		profile = await userprofile.load_profile(message.author.id)
```

Comment:
You're still querying the database for every message. This might also cause `profile` to point to the wrong profile in some edge cases.
Shouldn't this go at the beginning of the loop?

Reply:
fair, but i still think its wrong to allow users who haven't completed setup to interact with the bot. maybe we could just ignore their messages with the check function in `wait_for`, but i think that will still have the issue of querying on every message